ygot/diff.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2018 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ygot
+
+import (
+	"fmt"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/openconfig/ygot/util"
+
+	gnmipb "github.com/openconfig/gnmi/proto/gnmi"
+)
+
+// schemaPathTogNMIPath takes an input schema path represented as a slice of
+// strings, and returns a gNMI Path using the v0.4.0 path format Elem field
+// containing the elements. A schema path cannot specify any keys, and hence
+// only the path element's name is populated.
+func schemaPathTogNMIPath(path []string) *gnmipb.Path {
+	p := &gnmipb.Path{}
+	for _, pe := range path {
+		p.Elem = append(p.Elem, &gnmipb.PathElem{Name: pe})
+	}
+	return p
+}
+
+// joingNMIPaths takes a parent and child gNMI path, and concatenates the
+// child path to the parent path, returning the combined path. It only populates
+// the v0.4.0 Elem field.
+func joingNMIPaths(parent *gnmipb.Path, child *gnmipb.Path) *gnmipb.Path {
+	p := proto.Clone(parent).(*gnmipb.Path)
+	for _, e := range child.Elem {
+		p.Elem = append(p.Elem, e)
+	}
+	return p
+}
+
+// pathSpec is a wrapper type used to store a gNMI path for use as a map key.
+type pathSpec struct {
```

Comment:
I don't think I follow this comment. How does this struct help the gNMI path to become a map key? It still only contains a slice, i.e. not something truly hashable.

Or maybe I am misunderstanding it altogether?

Reply:
Yes, apologies -- this comment is wrong. The current design idea I have is that we'll have the map that is returned use pointers to this struct type for the key of the map. The `findSetLeaves` function in the other PR makes this clearer perhaps. I can update the comment there.